draft-bbs-signatures.md

```diff
@@ -291,6 +291,63 @@ For readability, this document makes these transformations implicitly, but they
 
 Optional input/parameters to operations that feature in a call to a HASH or XOF function, that are not supplied to the operation should default to an empty octet string. For example, if X is an optional input/parameter that is not supplied, whilst A and B are required, then the procedural step of `HASH(A || X || B)` MUST be evaluated to `HASH(A || "" || B)`.
 
+## Generator point computation
+
+The `compute_generators` function defines how to create a set of generators that form a part of the public parameters used by the BBS Signature scheme to accomplish operations such as Sign, Verify, ProofGen and ProofVerify. It takes one input, the maximum number of selectively revealed messages, which determines the number of generator points to create. The result is a sequence beginning with the `H_s` and `H_d` generators, followed by each of the `H_i` message generators.
+
+This document defines one implementation of this method, `compute_generators_xof`, which uses [@!I-D.irtf-cfrg-hash-to-curve] with an extensible-output hash function to generate a sequence of points with a uniformly random distribution from a seed value and DST.
+
+Each generator creation method must define a unique `GENERATORS_ID` identifier which is incorporated into the ciphersuite ID.
+
+### Security Considerations
+
+TODO
+
+### compute\_generators\_xof
+
+```
+generators = compute_generators_xof(maxMessageCount);
+
+Parameters:
+
+- hash_to_curve_g1: A cryptographic hash function that takes as an arbitrary octet string input and returns a point in G1 as defined in [@!I-D.irtf-cfrg-hash-to-curve]. The expand_message method MUST be expand_message_xof, and hash MUST be the same as xof.
+- generator_xof, an extensible-output cryptographic hash function.
+- generator_dst, octet string. The domain separation tag.
+- generator_seed, octet string. A unique seed value for the generator point sequence.
+
+Inputs:
+
+- maxMessageCount (REQUIRED), non-negative integer. The maximum number of selectively-revealed messages to be supported.
+
+Outputs:
+
+- generators, an array of generator points of length `maxMessageCount + 2`.
+
+Procedure:
+
+1. h = generator_xof(generator_seed)
+
+2. length = maxMessageCount + 2
```

Comment:
Just a though. It may be better to add the +2 when calling the function in the operations?? That way someone that is reading the procedures will not be confused why they got L+2 elements when they asked for L??